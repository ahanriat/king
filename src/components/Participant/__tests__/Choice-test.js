import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line

import ParticipantChoice from '../Choice';

// jest.enableAutomock();

describe('Participant Choice', () => {
  it('render', () => {
    const props = {
      value: 'Test String',
      onChange: () => {},
    };
    const output = renderer.create(
      <ParticipantChoice {...props} />
    );
    const tree = output.toJSON();
    expect(tree).toMatchSnapshot();

    const instance = output.getInstance();
    //
    // instance.handleInputChange('testPlayer');
    // expect(instance.state.currentInputText).toEqual('testPlayer');

    instance.state.currentInputText = 'testPlayer';

    instance.handleAddPlayer();
    expect(instance.state.selectedPlayers).toEqual(['testPlayer']);

    instance.handleRemovePlayer('testPlayer');
    expect(instance.state.selectedPlayers).toEqual([]);
  });
});
