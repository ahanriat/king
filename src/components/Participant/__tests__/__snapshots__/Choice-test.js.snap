exports[`Participant Choice render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column"
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start"
      }
    }>
    <TextInput
      autoFocus={true}
      onChangeText={[Function onChangeText]}
      onSubmitEditing={[Function onSubmitEditing]}
      returnKeyType="next"
      style={
        Array [
          Object {
            "backgroundColor": "#ff5555",
            "color": "#0000ff",
            "fontSize": 14,
            "height": 40,
            "minWidth": 60
          },
          Object {
            "width": 10
          }
        ]
      }
      value="" />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "backgroundColor": "#543212",
          "height": 40,
          "opacity": 1,
          "width": 40
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Add
      </Text>
    </View>
  </View>
</View>
`;
